compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1142,6 +1141,43 @@ class TreeUnpickler(reader: TastyReader,
         val denot = accessibleDenot(qual.tpe.widenIfUnstable, name, sig, target)
         makeSelect(qual, name, denot)
 
+      def readSelectIn(): Select =
+        var sname = readName()
+        val qual = readTerm()
+        val ownerTpe = readType()
+        val owner = ownerTpe.typeSymbol
+        val SignedName(name, sig, target) = sname: @unchecked // only methods with params use SELECTin
+        val qualType = qual.tpe.widenIfUnstable
+        val prefix = ctx.typeAssigner.maybeSkolemizePrefix(qualType, name)
+
+        /** Tasty should still be able to resolve a method from another root class,
+         *  even if it has been moved to a super type,
+         *  or an override has been removed.
+         *
+         *  This is tested in
+         *  - sbt-test/tasty-compat/remove-override
+         *  - sbt-test/tasty-compat/move-method
+         */
+        def lookupInSuper =
+          val cls = ownerTpe.classSymbol
+          if cls.exists then
+            cls.asClass.classDenot
+              .findMember(name, cls.thisType, EmptyFlags, excluded=Private)
+              .atSignature(sig, target)
+          else
+            NoDenotation
+
+        val denot =
+          val d = ownerTpe.decl(name).atSignature(sig, target)
+          (if !d.exists then lookupInSuper else d).asSeenFrom(prefix)
+
+        makeSelect(qual, name, denot)
+
+      def readSelectInPoly(): Select =
+        val tree = readSelectIn()
+        val info = readType()
+        tree.withType(tree.symbol.copy(info = info).termRef)
```

Comment:
I find it suspicious that we're creating "member" symbols on the fly like that, without adding them to their owner. I guess this is what nsc does?

Reply:
I'm not 100% sure because I can't tell where cloned symbols are added to their owners in nsc.  Our original idea was that we didn't want to run into overload resolution issues by entering it.